tests/ondemand/ondemand_number_in_string_tests.cpp

```diff
@@ -0,0 +1,122 @@
+#include "simdjson.h"
```

Comment:
Could you add tests where it should fail... e.g., cases where the value inside the string is not a valid number string?

So maybe something like this...

```C++
auto json = R"(["nan","infinity","1.0e"])"_padded;
ondemand::parser parser;
ondemand::document doc;
ASSERT_SUCCESS(parser.iterate(json).get(doc));
const std::string expected[3] = {"nan","infinity","1.0e"};
size_t pos{0};
for (auto value : doc) {
  double d;
  ASSERT_ERROR(value.get_double_in_string().get(d), simdjson::NUMBER_ERROR);
  std::string_view view;
  ASSERT_SUCCESS(value.get_string().get(view));
  ASSERT_EQUAL(expected, view);
  pos++;
 
}
ASSERT_EQUAL(pos,3);
```

(This code is probably not correct, it is just for illustration.)

Reply:
My proposed test also checks for something else: in case of error, you want to be able to fall back on checking the string itself... we want to make sure that works...